internal/fs/implicit_dirs_test.go

```diff
@@ -569,3 +569,64 @@ func (t *ImplicitDirsTest) AtimeCtimeAndMtime() {
 	ExpectThat(ctime, timeutil.TimeNear(mountTime, delta))
 	ExpectThat(mtime, timeutil.TimeNear(mountTime, delta))
 }
+
+// Create objects in implicit directories with
+// unsupported names such as ., .., /, \0 and
+// test that stat and ReadDirPicky on the different directories.
+func (t *ImplicitDirsTest) UnsupportedDirNames() {
+	var fi os.FileInfo
+	var entries []os.FileInfo
+	var err error
+
+	// Set up contents.
+	AssertEq(
```

Comment:
Let's use stretchr/testify package instead of "github.com/jacobsa/oglematchers" and "github.com/jacobsa/ogletest"

Reply:
This change will increase the size of this PR a lot . I'll take that up as a separate PR soon after.